src/kOS/Suffixed/VesselTarget.Parts.cs

```diff
@@ -0,0 +1,392 @@
+            allParts = null;
+            dockingPorts = null;
+            decouplers = null;
+            partCache = null;
+        }
+        public PartValue Root
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return rootPart;
+            }
+        }
+        public DecouplerValue NextDecoupler
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return nextDecoupler;
+            }
+        }
+        public ListValue<PartValue> Parts
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return allParts;
+            }
+        }
+        public ListValue<DockingPortValue> DockingPorts
+        {
+            get
+            {
+                if (dockingPorts == null)
+                    ConstructParts();
+                return dockingPorts;
+            }
+        }
+        public ListValue<DecouplerValue> Decouplers
+        {
+            get
+            {
+                if (decouplers == null)
+                    ConstructParts();
+                return decouplers;
+            }
+        }
+        public PartValue this[global::Part part]
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return partCache[part];
+            }
+        }
+        private void ConstructParts()
+        {
+            rootPart = null;
+            nextDecoupler = null;
+            allParts = new ListValue<PartValue>();
+            dockingPorts = new ListValue<DockingPortValue>();
+            decouplers = new ListValue<DecouplerValue>();
+            partCache = new Dictionary<global::Part, PartValue>();
+
+            ConstructPart(Vessel.rootPart, null, null);
+
+            allParts.IsReadOnly = true;
+            dockingPorts.IsReadOnly = true;
+            decouplers.IsReadOnly = true;
+        }
+        private void ConstructPart(global::Part part, PartValue parent, DecouplerValue decoupler)
+        {
+            if (part.State == PartStates.DEAD || part.transform == null)
+                return;
+
+            PartValue self = null;
+            foreach (var module in part.Modules)
+            {
+                if (module is IEngineStatus engine)
+                {
+                    self = new EngineValue(Shared, part, parent, decoupler);
+                    break;
+                }
+                if (module is IStageSeparator)
+                {
+                    if (module is ModuleDockingNode dock)
+                    {
+                        var port = new DockingPortValue(Shared, part, parent, decoupler, dock);
+                        self = port;
+                        dockingPorts.Add(port);
+                        if (dock.stagingEnabled)
```

Comment:
This would better be rewritten as
```
                        if (!dock.StagingEnabled())
                            break;
                        decoupler = port;
                        decouplers.Add(decoupler);
```
Mostly cosmetic, not changing the logic:
1. Rather `StagingEnabled()` instead of `stagingEnabled`
2. No need to check unchanged `decoupler` later, we can break here.